GoogleForCreators/web-stories-wp
# Templates: Add Puppeteer Script to Generate Template Page Screenshots

## Context

Following work done by @merapi this is the first step in getting template pages as images to render instead of Stories in the dashboard (as well as for page layouts in the editor).  

References: https://github.com/google/web-stories-wp/pull/8491, https://github.com/merapi/test-amp-screenshot/blob/master/index.js 

Thanks Marcin! This is such a cool idea, totally credit to you. 



## Summary

Adds script to templates package to run that will generate screenshots of every template page ( `npm run workflow:render-template-posters`).  
Screenshots added here: https://github.com/google/web-stories-wp/pull/8727

## Relevant Technical Choices

- Using emulateMediaFeatures to set 'prefers-reduced-motion' when looking at the preview of a template to prevent animations from running so that screenshots get all elements on the page. 
- Add a `slug` to each template and pass it through to use for asset naming so it's consistent. Added a test to make sure each template has one moving forward.

## To-do

- [x] Despite prefers-reduced-motion doing the bulk of the heavy lifting, there's a few template pages that don't look right. It might be just timing? 

- ~The Honeymooning in Italy template is broken. I filed a bug here. #8691  If you want to run the script locally to test it you'll need to comment out that template in `getTemplates` (`templateNames`).~
- We should add an option to run this for just one template so that it's less time consuming for templates added moving forward. This is my first time working with puppeteer like this though so I wanted to do it separately incase there's improvements I can make here that would come in handy with that. 

## User-facing changes

None

## Testing Instructions

Try running the script locally. If you don't want to watch this thing for an hour, just set the `templateCount` to like 3 and you can see it run. 

### QA






This PR can be tested by following these steps:

1.

### UAT







This PR can be tested by following these steps:

1.

## Reviews

### Does this PR have a security-related impact?

No

### Does this PR change what data or activity we track or use?

No

### Does this PR have a legal-related impact?

No

## Checklist




---



Partially addresses #8373
